tests/Doctrine/Tests/ORM/Query/SelectSqlMappingTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Query;
+
+use Doctrine\DBAL\Types\FloatType;
+use Doctrine\DBAL\Types\IntegerType;
+use Doctrine\ORM\Query as ORMQuery;
+use Doctrine\Tests\OrmTestCase;
+use Exception;
+use ReflectionObject;
+use function array_search;
+use function sprintf;
+
+class SelectSqlMappingTest extends OrmTestCase
+{
+    private $em;
+
+    protected function setUp() : void
+    {
+        $this->em = $this->getTestEntityManager();
+    }
+
+    /**
+     * Assert a valid SQL generation.
+     *
+     * @param string $dqlToBeTested
+     * @param array  $types
+     */
+    public function assertScalarTypes($dqlToBeTested, array $types)
+    {
+        try {
+            $query = $this->em->createQuery($dqlToBeTested);
+
+            $query
+                ->setHint(ORMQuery::HINT_FORCE_PARTIAL_LOAD, true)
+                ->useQueryCache(false);
+
+            $r      = new ReflectionObject($query);
+            $method = $r->getMethod('getResultSetMapping');
+            $method->setAccessible(true);
+            /** @var ORMQuery\ResultSetMapping $mapping */
+            $mapping = $method->invoke($query);
+            foreach ($types as $key => $expectedType) {
+                $alias = array_search($key, $mapping->scalarMappings);
+                self::assertInstanceOf(
+                    $expectedType,
+                    $mapping->typeMappings[$alias],
+                    sprintf('The field "%s" was expected as a $expectedType', $key)
```

Comment:
`$expectedType` value would not be printed, you should use another `%s` parameter